consumer/src/main/java/com/flipkart/varadhi/consumer/VaradhiConsumer.java

```diff
@@ -0,0 +1,38 @@
+package com.flipkart.varadhi.consumer;
+
+import com.flipkart.varadhi.entities.VaradhiSubscription;
+
+/**
+ * Base varadhi consumer to consume messages from a varadhi topic. It corresponds to a single shard of some
+ * subscription.
+ */
+public interface VaradhiConsumer {
+
+    VaradhiSubscription getSubscription();
```

Comment:
ya. this will be reworked to point to a shard. but getter will be there. it is a consumer for a subscription, so a pure getter to again fetch the subscirption details for whcih this consumer was created seems useful. Also, the same can probably be used to check whether the consumer is created for a subscription with an old version.

If it turns out that the this method is not useful, we can remove it.